external-plugins/test-subset/plugin/handler/handler.go

```diff
@@ -0,0 +1,363 @@
+		var event github.IssueCommentEvent
+		if err := json.Unmarshal(incomingEvent.Payload, &event); err != nil {
+			log.WithError(err).Error("Could not unmarshal event.")
+			return
+		}
+		h.handlePullRequestEvent(eventLog, &event)
+	default:
+		log.Infoln("Dropping irrelevant:", incomingEvent.Type, incomingEvent.GUID)
+	}
+}
+
+// For unit tests, as we create a local git NewFakeClient
+func (h *GitHubEventsHandler) SetLocalConfLoad() {
+	LoadConfigBytesFunc = loadLocalConfigBytes
+}
+
+func (h *GitHubEventsHandler) handlePullRequestEvent(log *logrus.Entry, event *github.IssueCommentEvent) {
+	if !h.shouldActOnIssueComment(event) {
+		return
+	}
+
+	org, repo := OrgRepo(event.Repo.FullName)
+	if !(org == "kubevirt" && repo == "kubevirt") {
+		return
+	}
+
+	pr, err := h.ghClient.GetPullRequest(org, repo, event.Issue.Number)
+	if err != nil {
+		log.WithError(err).Errorf("Could not get PR number %d", event.Issue.Number)
+		return
+	}
+
+	if !isPRReady(*pr) {
+		return
+	}
+
+	if !h.canUserTrigger(org, event.Comment.User.Login) {
+		return
+	}
+
+	testSubsetCommentRe := regexp.MustCompile(`^/test-subset (\S+) (\S+|\(.*\))$`)
+	if !testSubsetCommentRe.MatchString(event.Comment.Body) {
+		log.Errorf("Comment does not match the expected syntax.")
+		return
+	}
+
+	matches := testSubsetCommentRe.FindStringSubmatch(event.Comment.Body)
+	// extra precaution, not mandatory due to the regex
+	if len(matches) < 3 {
+		return
+	}
+
+	jobName := matches[1]
+	labels := matches[2]
+	if !strings.HasPrefix(labels, "(") {
+		labels = "(" + labels + ")"
+	}
+
+	presumbits, err := h.loadPresubmits(*pr)
+	if err != nil {
+		log.WithError(err).Errorf("loadPresubmits failed")
+		return
+	}
+
+	if presumbits == nil {
+		return
+	}
+
+	var presubmit config.Presubmit
+	for _, p := range presumbits {
+		if p.Name != jobName {
+			continue
+		}
+		presubmit = p
+		break
+	}
+
+	if presubmit.Name == "" {
+		return
+	}
+
+	presubmit.Spec.Containers[0].Env = append(presubmit.Spec.Containers[0].Env,
+		k8s_v1.EnvVar{Name: "KUBEVIRT_LABEL_FILTER", Value: labels})
+
+	job := pjutil.NewPresubmit(*pr, pr.Base.SHA, presubmit, event.GUID, map[string]string{})
+	if job.Labels == nil {
+		job.Labels = make(map[string]string)
+	}
+	job.Labels["prow-test-subset"] = "true"
+	testSubsetName := strings.Join([]string{"test-subset", job.Spec.Job}, "-")
+	job.Spec.Job = testSubsetName
+	job.Spec.Context = testSubsetName
+	_, err = h.prowClient.Create(context.Background(), &job, metav1.CreateOptions{})
+	if err != nil {
+		log.WithError(err).Errorf("Error creating prow job")
+		return
+	}
+}
+
+func (h *GitHubEventsHandler) loadPresubmits(pr github.PullRequest) ([]config.Presubmit, error) {
+	if pr.Base.Ref != "main" && pr.Base.Ref != "master" {
+		return nil, nil
+	}
+
+	pc, err := h.loadProwConfig(pr.Base.Repo.FullName)
+	if err != nil {
+		log.WithError(err).Errorf("Could not load prow config")
+		return nil, err
+	}
+
+	org, repo := OrgRepo(pr.Base.Repo.FullName)
```

Comment:
btw here i implemented the [OrgRepo](https://github.com/kubevirt/project-infra/pull/3164/files#diff-8383ef4cc8ae969cfad9de54ad4822aefef45adc934082855cc975a25d057d24R357) because the old functioned with that name doesn't exists
so i assume pr.Base.Repo.FullName does equal "kubevirt/kubevirt"

Reply:
Not sure, do you mean this one? https://github.com/kubevirt/project-infra/blob/0c50140b09af8beffb92c27f8f9319f33f851af3/robots/pkg/github/events.go#L28

Reply:
admitttedly a bit hidden in the events.go file

Reply:
thanks, seems so yep i can try to use that
ps.
https://github.com/kubevirt/project-infra/blob/0c50140b09af8beffb92c27f8f9319f33f851af3/robots/pkg/github/events.go#L30
aint it a bug ? i.e we first need to assert length of the split array and then access it, else it can panic

Reply:
yeah would be safer if that was done first, agree